isisd/isis_pdu.c

```diff
@@ -2153,9 +2153,6 @@ isis_receive (struct thread *thread)
 
   retval = circuit->rx (circuit, ssnpa);
 
-  if (retval == ISIS_OK)
-    retval = isis_handle_pdu (circuit, ssnpa);
-
```

Comment:
I think this will break all platforms other than those using BPF, since `isis_handle_pdu` won't be called for them anymore.

Reply:
I think isisd ONLY use bpf. What other platforms?

Reply:
The platform FRR supports most is Linux, that is why we have a lot of tools to get errors there and the reason why BPF was broken in the first place. Linux uses PF_PACKET (the socket approach) instead of the *BSD device approach, hence your confusion here about the "BPF" terminology.

BPF stands for Berkeley Packet Filter which defines a packet filtering instructions independently. On *BSDs you have to configure a "device" which is found on `/dev/bpf*` to use it, meanwhile in Linux we can apply this filter to an existing socket.

You can find more technical information in the `isis_pfpacket.c` and `isis_bpf.c` files, but the point is: we say *BSDs use the BPF method, because they require this special unix device to use BPF, meanwhile in Linux you can apply to a socket and so it is not explicit (more generic).

Here are the supported methods selection in case you are curious about what each OS uses:
https://github.com/FRRouting/frr/blob/master/configure.ac#L1102